### PR TITLE
Pyguide: consistent region added, streamsx.standard.files.CSVReader sample added

### PR DESCRIPTION
**Pull request by markheger**
Updates in python-appapi-devguide-4 and python-appapi-devguide-5

#188

### CODE REVIEW

File: docs/python/1.6/python-appapi-devguide-4.md [hunk 1]
```diff
@@ -378,10 +378,68 @@ Sample output:
...
+Another option to work with files in your streaming application is offered by the Python module [streamsx.standard](https://streamsxstandard.readthedocs.io/en/latest/generated/streamsx.standard.files.html#module-streamsx.standard.files).
+
+In the following sample, the [streamsx.standard.files.CSVReader](https://streamsxstandard.readthedocs.io/en/latest/generated/streamsx.standard.files.html#streamsx.standard.files.CSVReader) is used to read a file in the source callable. For this, we import the python module like below:
+
+~~~~~ python
+import streamsx.standard.files as files
+~~~~~
+
+The [streamsx.standard.files.CSVReader](https://streamsxstandard.readthedocs.io/en/latest/generated/streamsx.standard.files.html#streamsx.standard.files.CSVReader) accepts the file parameter either set as relative path to application directory or as absolute path.
+In this sample the file is added application bundle and the Streams application reads this file at runtime from the application directory.
```
Comment (ghost) [inline docs/python/1.6/python-appapi-devguide-4.md]: In this sample, the file is added to the application bundle ...
"'to the" is missing.

File: docs/python/1.6/python-appapi-devguide-5.md [hunk 2]
```diff
@@ -67,3 +77,181 @@ def main():
...
+<a id="consistent_region"></a>
+
+# Consistent region
+
+Because of business requirements, some applications require that all tuples in an application are processed at least once.
+By default any state is reset to its initial state after a processing element (PE) restart. A restart may occur due to:
+
+* a failure in the PE or its resource,
+* a explicit PE restart request,
+* a parallel region width change (IBM Streams 4.3 or later)
```
Comment (ghost) [inline docs/python/1.6/python-appapi-devguide-5.md]: A parallel region width change is not supported by Streams when the parallel region belongs to a consistent region. Therefore, I would NOT mention the third bullet of the list here. For autonomous regions, this bullet is correct.

Comment (markheger) [reply docs/python/1.6/python-appapi-devguide-5.md]: should be removed

File: docs/python/1.6/python-appapi-devguide-5.md [hunk 3]
```diff
@@ -67,3 +77,181 @@ def main():
...
+<a id="configure"></a>
+## Configure a consistent region
+
+* [streamsx.topology.topology.Stream.set_consistent()](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.topology.html#streamsx.topology.topology.Stream.set_consistent)
+* [ConsistentRegionConfig](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.state.html#streamsx.topology.state.ConsistentRegionConfig)
+
+The recommended way to create a `ConsistentRegionConfig` is to call either `operator_driven()` or `periodic()`.
+
+For example, set a source stream `s` to be a the start of an operator driven consistent region with a drain timeout of five seconds and a reset timeout of twenty seconds:
+```
```
Comment (ghost) [inline docs/python/1.6/python-appapi-devguide-5.md:124]: Drain and reset timeouts of 5 or 20 seconds are no good examples. Readers might think, that these values are good values, which fit the average streaming application in most cases. The default timeouts for drain/reset are 180 seconds, I think. This example should go to higher timeouts, at least 30 or 60 seconds.

File: docs/python/1.6/python-appapi-devguide-5.md [hunk 4]
```diff
@@ -67,3 +77,181 @@ def main():
...
+* [Use SPL operators supporting consistent region](#spl_cr)
+
+
+<a id="configure"></a>
+## Configure a consistent region
+
+* [streamsx.topology.topology.Stream.set_consistent()](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.topology.html#streamsx.topology.topology.Stream.set_consistent)
+* [ConsistentRegionConfig](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.state.html#streamsx.topology.state.ConsistentRegionConfig)
+
+The recommended way to create a `ConsistentRegionConfig` is to call either `operator_driven()` or `periodic()`.
```
Comment (ghost) [inline docs/python/1.6/python-appapi-devguide-5.md:116]: It's worth to mention, that `operator_driven()` cannot be used when a Python callable is the begin of the CR as there is no Python API to trigger the region as we have for C++ or Java SPL operators. When SPL operators (Java or C++) are the start of a CR, the operator dictates when the region is made consistent, i.e. drained and checkpointed. Often additional operator parameters are required to control when the operator is supposed to trigger the region becoming consistent.

File: docs/python/1.6/python-appapi-devguide-5.md [hunk 5]
```diff
@@ -67,3 +77,181 @@ def main():
...
+
+Reference
+* [streamsx.topology.state](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.state.html#module-streamsx.topology.state)
+
+<a id="stateful"></a>
+## Stateful callables
+
+Use of a class instance allows a transformation to be stateful by maintaining state in instance attributes across invocations.
+
+When the callable is in a consistent region then it is serialized using dill.
```
Comment (ghost) [inline docs/python/1.6/python-appapi-devguide-5.md]: omit the word "then" in the sentence _When the callable is in a consistent region then it is serialized using dill._

File: docs/python/1.6/python-appapi-devguide.md [hunk 6]
```diff
@@ -46,8 +46,7 @@ Applications created with the IBM Streams Python API are executed on an instance
 
  - Use the **Streaming Analytics service running on IBM Cloud**: The Streaming Analytics service is a cloud version of IBM Streams, so you don't need to install Streams to build Python applications for the service. [Create a free instance of the Streaming Analytics service here](https://cloud.ibm.com/catalog/services/streaming-analytics).  The applications you create will run in the IBM Cloud.
 
-
- -  Enable the **IBM Streams add-on in IBM Cloud Pak for Data**: IBM Streams is included as an add-on in for IBM Cloud Pak for Data. Contact your administrator to enable the add-on.  Streams can also be installed as a stand-alone deployment on Red Hat OpenShift or Kubernetes environments.  
+ -  Enable the **IBM Streams add-on in IBM Cloud Pak for Data**: IBM Streams is included as an add-on in for IBM Cloud Pak for Data. Contact your administrator to enable the add-on.  Streams can also be installed as a stand-alone deployment on Red Hat OpenShift or Kubernetes environments.
```
Comment (ghost) [inline docs/python/1.6/python-appapi-devguide.md]: _IBM Streams is included as an add-on in for IBM Cloud Pak for Data._ Remove the "in" after add-on.

File: docs/python/1.6/python-appapi-devguide-5.md (hunk 4 above)
Comment (markheger) [reply docs/python/1.6/python-appapi-devguide-5.md:116]: note box needs to be added

**Pull request merged**